test/multinomial.lua

```diff
@@ -0,0 +1,52 @@
+local tester = torch.Tester()
+
+cmd = torch.CmdLine()
+cmd:text()
+cmd:text()
+cmd:text('Testing alias multinomial on cuda')
+cmd:text()
+cmd:text('Options')
+cmd:option('--compare',false,'compare with cutorch multinomial')
+cmd:text()
+
+-- parse input params
+params = cmd:parse(arg)
+
+require 'cutorch'
+local function checkMultinomial()
+   local n_class = {10, 100, 1000}
+   local n_sample = {10, 100, 1000, 10000}
+   local n_dist = 100
+   for _, curr_n_class in pairs(n_class) do
+      for _, curr_n_sample in pairs(n_sample) do
+	 print("")
+	 print("Benchmarking multinomial with "..curr_n_class.." classes and "..curr_n_sample.." samples")
+	 torch.seed()
+	 local probs = torch.CudaDoubleTensor(n_dist, curr_n_class):uniform(0,1)
+	 local a = torch.Timer()
+	 local cold_time = a:time().real
+	 a:reset()
+	 cutorch.synchronize()
+	 a:reset()
+	 for i = 1,10 do
+	    torch.multinomial(probs, curr_n_sample, true)
+	    cutorch.synchronize()
```

Comment:
why are you synchronizing every time through for the benchmark? one should only synchronize at the beginning and the end.